File: AJM/DAY6/11055.py
#가장 큰 증가 부분 수열


"""
0  1   2  3  4  5  6  7  8  9
10 100 2  50 60 3  5  6  7  8

dp
1 0   0
1 101
1 101 3
1 101 3 53

"""
n = int(input())
arr = list(map(int,input().split()))

dp = [0]*1000
dp[0] = arr[0]

for i in range(1,n):
    for j in range(i):
        if arr[i] > arr[j]:
            dp[i] = max(dp[i],dp[j]+arr[i])
        else:
            dp[i] = max(dp[i],arr[i])
    
print(max(dp))

File: MY/DAY5/11055.java
/*
 * @author Minyeong Park
 * @date 2023.02.13.
 * 가장 큰 증가 부분 수열 (Bottom up)
 * 반례 참고 : https://www.acmicpc.net/board/view/87941
 * 문제 링크 : https://www.acmicpc.net/problem/11055
 */



import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;
        int n = Integer.parseInt(br.readLine());
        int[] arr = new int[n];
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }
        int[] dp = new int[n];
        dp[0] = arr[0];

        for (int i = 1; i < n; i++) {
            dp[i] = arr[i]; // i번째 보다 작은 인덱스를 탐색할 때 먼저 dp[i]를 초기화해주는 것 잊지 말자!
            for (int j = 0; j < i; j++) {
                if (arr[j] < arr[i]) {
                    dp[i] = Math.max(dp[i], dp[j] + arr[i]);
                }
            }
        }

        int max = 0;
        for (int sum : dp) {
            max = Math.max(sum, max);
        }

        System.out.println(max);
    }
}